lib/testing/matrix.js

```diff
@@ -0,0 +1,171 @@
+'use strict'; // eslint-disable-line strict
+
+/**
+* Running tests with multiple configurations using varying parameters
+* @constructor
+*/
+class TestMatrix {
+    /**
+    * Builds a new testing Matrix
+    *
+    * @param {Object.<string, string|string[]>} params - Argument of the matrix
+    *
+    * @returns {TestMatrix} new testing Matrix
+    */
+    constructor(params) {
+        /**
+        * Create deep copy for forbidden the user to modify the object outside
+        * the class
+        **/
+        this.params = Object.assign({}, params);
+        this.paramsOnString = JSON.stringify(this.params);
```

Comment:
Why not initialize the object here and do the stringify later in the `toString()` method?